jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -582,6 +582,13 @@ private static void ensureProperties(boolean isInitialization) {
 	initializedProperties.put("ibm.system.encoding", platformEncoding); //$NON-NLS-1$
 	initializedProperties.put("sun.jnu.encoding", platformEncoding); //$NON-NLS-1$
 	initializedProperties.put("file.encoding", fileEncoding); //$NON-NLS-1$
+	/*[IF JAVA_SPEC_VERSION >= 17]*/
+	if (fileEncoding == null) {
+		initializedProperties.put("native.encoding", platformEncoding); //$NON-NLS-1$
```

Comment:
fyi the change in https://github.com/eclipse-openj9/openj9/pull/12642, which uses `(fileEncoding != null) ? fileEncoding : platformEncoding`, which is a little more compact.

Reply:
Updated.